glusterd2/xlator/options/options.go

```diff
@@ -254,21 +260,37 @@ func ValidatePriorityOrSize(o *Option, val string) error {
 
 // ValidateSizeList validates if the option is a valid size list
 func ValidateSizeList(o *Option, val string) error {
-	return ErrInvalidArg
+	var sizeinbytes int64
+	slist := strings.Split(val, ",")
+	for _, el := range slist {
+		l := len(el)
+		if strings.ContainsRune(el, 'B') || strings.ContainsRune(el, 'b') {
+			size := el[l-2 : l]
+			t := strings.TrimSpace(el[:l-2])
+			v, err := strconv.ParseInt(t, 10, 64)
+			if err != nil {
+				return err
+			}
+			switch size {
+			case "KB", "kb":
+				sizeinbytes = v * 1000
```

Comment:
This should be 1024, right ?

Reply:
@prashanthpai am not sure. I read it up, I think 1KB=1000bytes and 1KiB=1024bytes.